src/main/java/org/spongepowered/api/map/color/MapColor.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * This file is part of SpongeAPI, licensed under the MIT License (MIT).
+ *
+ * Copyright (c) SpongePowered <https://www.spongepowered.org>
+ * Copyright (c) contributors
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.spongepowered.api.map.color;
+
+import com.google.common.base.Optional;
+import org.spongepowered.api.data.DataSerializable;
+import org.spongepowered.api.util.Color;
+
+public interface MapColor extends DataSerializable {
```

Comment:
Because this is a lot of hassle for a plugin to get all the colors to
perform custom color matching in a plugin. So a helper is still needed to
get the whole value list.

On Sat, Apr 2, 2016, 3:18 PM Walker Crouse <notifications@github.com> wrote:

> In src/main/java/org/spongepowered/api/map/color/MapColor.java
> <https://github.com/SpongePowered/SpongeAPI/pull/1162#discussion_r58299255>
> :
>
> > + *
> > + * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
> > + * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
> > + * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
> > + * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
> > + * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
> > + * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
> > + * THE SOFTWARE.
> > + */
> > +package org.spongepowered.api.map.color;
> > +
> > +import com.google.common.base.Optional;
> > +import org.spongepowered.api.data.DataSerializable;
> > +import org.spongepowered.api.util.Color;
> > +
> > +public interface MapColor extends DataSerializable {
>
> How so? It could potentially look something like
>
> MapColors.GRASS.darker();// orMapColors.GRASS.shade(MapShades.DARK);
>
> How is that confusing? You could even add a helper to MapColor if you
> wanted
>
> MapColor.of(MapColors.GRASS, MapShades.DARKER);
>
> —
> You are receiving this because you authored the thread.
>
> Reply to this email directly or view it on GitHub
> <https://github.com/SpongePowered/SpongeAPI/pull/1162/files/d8338269be828e278d15ed5ffbe9bb30b7c899c1#r58299255>
>